### PR TITLE
Add ROI centroid flag

### PR DESCRIPTION
**Pull request by CodyCBakerPhD**
First modularization of #117

### CODE REVIEW

File: src/neuroconv/tools/roiextractors/roiextractors.py [hunk 1]
```diff
@@ -671,9 +674,14 @@ def image_mask_iterator():
             ],
             id=roi_ids,
         )
-
-        image_segmentation.create_plane_segmentation(**plane_segmentation_kwargs)
-
+        plane_segmentation = PlaneSegmentation(**plane_segmentation_kwargs)
+        if include_roi_centroids:
+            tranpose_roi_location_axes = (1, 0) if len(segmentation_extractor.get_image_size()) == 2 else (1, 0, 2)
```
Comment (h-mayorquin) [inline src/neuroconv/tools/roiextractors/roiextractors.py:679]: Can you write a brief comment on which case each of this if covers. Or maybe an appropriate name for the boolean variable instead? `dealing_with_non_volumetric_data = len(segmentation_extractor.get_image_size())==2` or something like this?

I just think that it took me a minute or two what this was for and I think I will forget in the future.

File: src/neuroconv/tools/roiextractors/roiextractors.py [hunk 2]
```diff
@@ -671,9 +674,14 @@ def image_mask_iterator():
...
+            roi_locations = segmentation_extractor.get_roi_locations()[tranpose_roi_location_axes, :].T
```
Comment (h-mayorquin) [inline src/neuroconv/tools/roiextractors/roiextractors.py:680]: I was kind of confused about the double transpose (as another transpose is repeated on line 680) at first glance. Maybe some comment about that? 

I get that you require to write the images in the expected format (columns vs heights) [hence transpose in lines 677] and then you transpose all the roi_locations to get num_rois in the first dimension. Not related tot his PR but should not we return the roi locations in shape `(num_rois, shape_of_the_images)`?

**Review by h-mayorquin — CHANGES_REQUESTED** [review]
Looks good to me. Thanks for breaking this apart. Some minimal request for the sake of future readability of the code.

File: src/neuroconv/tools/roiextractors/roiextractors.py (hunk 1 above)
Comment (CodyCBakerPhD) [reply src/neuroconv/tools/roiextractors/roiextractors.py]: Added comment

File: src/neuroconv/tools/roiextractors/roiextractors.py (hunk 2 above)
Comment (CodyCBakerPhD) [reply src/neuroconv/tools/roiextractors/roiextractors.py]: Moved transpose to emphasize connection to writing as column to table

**Review by h-mayorquin — COMMENTED** [review]
There are some on-going conflicts with this one.

**Review by h-mayorquin — APPROVED** [review]

**Pull request merged**